employee/sealed.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Starting from a `read_only` struct, this program
+ * restores full privileges using a previous `sealed`
+ * capabilty.
+ */
+
+#include "include/employee.h"
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/sysctl.h>
+
+
+
+int main()
+{
+	struct employee employee = {0, "Good", "Employee", 55000};
+	struct employee *ro_employee = set_read_only(&employee);
+	printf("*** `ro_employee` is read-only ***");
+	pp_cap(ro_employee);
+	print_details(ro_employee);
+	// Data to update the employee
+	struct employee updated_employee = {1, "Better", "Unsealed", 250000};
+	struct employee *updated_employee_ptr = &updated_employee;
+	// Pointers for the sealed and unsealed capability
+	struct employee *sealed_employee;
```

Comment:
Consider moving declarations to top or declaring at point of first assignment.

Reply:
Oh, sure.

Reply:
We're always in C99-or-later land, so "declare at first assignment" is generally the way to go (though, being C, there are sometimes reasons why that can't be done).

Reply:
I think you both have a good point, being C or C-99. I honestly think I have just followed the example code here :) Anyway, "declare at first assignment" seems reasonable in 2021, I guess.